pkgs/io_file/lib/windows_file_system.dart

```diff
@@ -0,0 +1,6 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+export 'src/vm_windows_file_system.dart'
+    if (dart.library.html) 'src/web_windows_file_system.dart';
```

Comment:
I'm curious - the web file systems will have differences between the OS platforms?

Reply:
I'm not sure what API design is right but I want the user to be able to write something like:

```dart
bool isHidden(String path) {
  if (fileSystem is WindowsFileSystem) {
    return fileSystem.metadata(path).isHidden;
  } else {
    return path.startsWith(".");
  }
}
```

Either `package:io_file` has to protect the web developer from `ffi` imports or the developer will have to do that themselves (like they do have to with `package:http`: https://github.com/dart-lang/http/blob/9129a96be871a5779436f042bd13790960171373/pkgs/flutter_http_example/lib/main.dart#L13).